classes/notification/PKPNotificationOperationManager.inc.php

```diff
@@ -367,15 +367,21 @@ private function formatNotification($request, $notification, $notificationTempla
 	 * Send an email to a user regarding the notification
 	 * @param $request PKPRequest
 	 * @param $notification object Notification
+	 * @param $mailConfigurator callable If specified, must return a MailTemplate instance. A ready TemplateMail object will be provided as argument
 	 */
-	private function sendNotificationEmail($request, $notification) {
+	private function sendNotificationEmail(Request $request, Notification $notification, $contextId, callable $mailConfigurator) {
 		$userId = $notification->getUserId();
 		$userDao = DAORegistry::getDAO('UserDAO');
 		$user = $userDao->getById($userId);
 		if ($user && !$user->getDisabled()) {
 			AppLocale::requireComponents(LOCALE_COMPONENT_APP_COMMON);
 
 			$context = $request->getContext();
+			if ($contextId && (!$context || $context->getId() != $contextId)) {
+				$contextDao = Application::getContextDAO();
```

Comment:
If I remember well, I've added this because I wanted to use a specific context from the command-line/scheduler.